src/mqtt/qmqtt_client_p.cpp

```diff
@@ -206,10 +206,9 @@ void QMQTT::ClientPrivate::sendConnect()
     if(!willTopic().isEmpty())
     {
         frame.writeString(willTopic());
-        if(!willMessage().isEmpty())
-        {
-            frame.writeByteArray(_willMessage);
-        }
+        // Transmit last will message even if it is empty, otherwise an invalid connect frame is send (user name is 
```

Comment:
actually, there is no need in a descriptive comment. passing willTopic without willMessage is obviously a bug (3.1.3.2 and 3.1.3.3 at http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718031)

Reply:
From my point of view it is not a bug. Section 3.1.3.3 explicitly mention a zero length will message: "This field consists of a two byte length followed by the payload for the Will Message expressed as a sequence of **zero** or more bytes."

With the original implementation it is not possible to connect to a mqtt server with a will topic and an empty will message. With the proposed change everything works as expected.

If you do not want the descriptive comment, please delete it.